src/components/SingleArticle.jsx

```diff
@@ -10,26 +10,41 @@ export default function SingleArticle() {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
   const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
   const [votes, setVotes] = useState(0);
 
   useEffect(() => {
     getArticleById(article_id).then((response) => {
       setArticle(response.data.article);
       setVotes(response.data.article.votes);
       setIsLoading(false);
+      setError(null);
     });
   }, []);
 
   const handleUpvote = () => {
     setVotes(votes + 1);
-    patchArticleUpvote(article_id);
+    patchArticleUpvote(article_id)
+      .then((response) => {
+        setError(null);
+      })
+      .catch((err) => {
+        setError("Voting failed, please try again later");
```

Comment:
Nice job with displaying an error, but the votes stay increased/decreased on the page. You can reverse the setVotes function in the catch block so it goes back to the original vote count pre-vote :)